pintc/tests/cli.rs

```diff
@@ -142,8 +156,17 @@ fn explicit_output() {
     assert!(output_file.exists());
     let _ = fs::remove_file(output_file);
 
+    let input_file_name = input_file.path().file_name().unwrap().to_str().unwrap();
+    let mut expected_output_file = tempfile::NamedTempFile::new().unwrap();
+    let expected_output = format!("    contract {input_file_name}       EB87FCE275A9AB10996D212F39221A56B90E01C37FA9D16EE04A3FE8E17DEED9
+         └── {input_file_name}::test BA6595C5C75346E6C82BED0CE770D0758ADD1712163FCE45E38E5E8EAC6AA153\n");
+    write!(expected_output_file.as_file_mut(), "{}", expected_output).unwrap();
+
     check(&output.stderr, expect_test::expect![""]);
-    check(&output.stdout, expect_test::expect![""]);
+    check_file(
+        &output.stdout,
+        expect_test::expect_file![expected_output_file.path()],
+    );
```

Comment:
Super ugly. It was the only way I could get the tests to work properly while still allowing `trunk fmt` to run. I'd like it to match the output directly like 
```
r#"
    contract {input_file_name}  EB87FCE275A9AB10996D212F39221A56B90E01C37FA9D16EE04A3FE8E17DEED9
    └── {input_file_name}::test BA6595C5C75346E6C82BED0CE770D0758ADD1712163FCE45E38E5E8EAC6AA153
"#
```
Which I can't do if I want to write to a file.

I also had to write to a file because the `expect![` macro was removing all whitespace at the beginning of the `string literal`, even when it was a `raw` string. It also meant I couldn't use the dynamic tempfile filename. Hardcoding it wasn't an issue, it just meant that it was more to deal with when creating a test.

Open to ideas here. Spent a few hours on it and have decided that life wasn't meant to be perfect.

Reply:
Does it absolutely, truly matter if the whitespace is a little off in the test?  What if, as a hack for convenience, we use the `expect!` as usual, but throw in an ancillary `assert_eq!(output[0..4], "    ")` or whatever, which forever ensures the whitespace is there, pinky promise.